server/src/main/java/org/opensearch/index/mapper/IpFieldMapper.java

```diff
@@ -263,29 +269,66 @@ public Query termQuery(Object value, @Nullable QueryShardContext context) {
             return query;
         }
 
+        // @TODO check strings, byterefs, inetaddresses for concrete and masks
         @Override
         public Query termsQuery(List<?> values, QueryShardContext context) {
             failIfNotIndexedAndNoDocValues();
-            InetAddress[] addresses = new InetAddress[values.size()];
-            int i = 0;
-            for (Object value : values) {
-                InetAddress address;
+            List<InetAddress> concreteIPs = new ArrayList<>();
+            List<Query> ranges = new ArrayList<>();
+            IpMultiRangeQueryBuilder multiRange = new IpMultiRangeQueryBuilder(name());
```

Comment:
MultiRangeQuery handles points index, but there's no a special case for DV only field.